### PR TITLE
CSS classes by BEM, Added 1x1 gif placeholder image, and more...

### PR DESCRIPTION
**Pull request by RuslanZavacky**
Hey,

I've did a lot of changes, seems like they might be useful and I hope you will accept them :)

**Changelog**
- [x] CSS classes by BEM. So they can be easily rewritable by other applications.
- [x] Added 1x1 gif placeholder image to avoid "broken" images
- [x] Fixed CSS styling for center positioning. Center positioning looked strange.
- [x] Removed events when they became obsolete. When image is already lazy loaded, no need to handle scroll event and also image load/error events.
- [x] Added threshold for image in viewport detection. Its better when you can control when you can start to load images when user is scrolling to them.

### CODE REVIEW

**Comment by twokul** [comment]
@RuslanZavacky thanks for the patch dude! I will review tonight. Tests seem to be failing.

**Comment by RuslanZavacky** [comment]
@twokul yeap, strange. ember test --server tests passes, ember test return nothing for me, like no tests :) I'll dig into it more, if you find something in PR, comment and I'll fix it :)

**Comment by RuslanZavacky** [comment]
@twokul I've fixed tests :)

**Comment by twokul** [comment]
@RuslanZavacky thanks dude! I'll review closely tomorrow

File: addon/components/lazy-image.js [hunk 1]
```diff
@@ -6,17 +6,20 @@ var get       = Ember.get;
 var set       = Ember.set;
 var observer  = Ember.observer;
 var Component = Ember.Component;
+var defaultLazyImage = "assets/lz-image/lazy_1x1.gif";
```
Comment (twokul) [inline addon/components/lazy-image.js]: I would avoid adding assets tbh. I'd rather have a placeholder svg

File: README.md [hunk 2]
```diff
@@ -41,6 +41,12 @@ You can customize `error` text by passing it as an parameter:
 {{lazy-image url='http://my-not-valid-url.com/foo.jpg' errorText='Something went wrong.'}}
 ```
 
+Also it is possible to define threshold (px) for image in viewport detection, so image is preloaded when user is scrolling to it:
+
+```html
+{{lazy-image url='http://my-not-valid-url.com/foo.jpg' threshold=300}}
```
Comment (twokul) [inline README.md]: this seems cool

File: addon/mixins/in-viewport.js [hunk 3]
```diff
@@ -3,53 +3,62 @@
 
 import Ember from 'ember';
 
-var on           = Ember.on;
-var get          = Ember.get;
-var Mixin        = Ember.Mixin;
-var debounce     = Ember.run.debounce;
+var on = Ember.on;
```
Comment (twokul) [inline addon/mixins/in-viewport.js]: this PR is very hard to read b/c you change the formatting. do you mind leaving the formatting as it was?

Comment (RuslanZavacky) [inline addon/mixins/in-viewport.js]: I've reformatted only variables at the beginning of the file. But when you format variable by the equal sign, this is the price you pay, when you add a new variable. Then you have to reformat whole column.  Same goes for formatting by : sign. when you add variable with the longer name then you again will have a PR where whole column is changed. If you insist, I can change it.

File: addon/components/lazy-image.js (hunk 1 above)
Comment (RuslanZavacky) [inline addon/components/lazy-image.js]: Yea, I also dislike this approach, but do you have experience with using SVG as a placeholder? How Chrome will react on it? Because when chrome does not have an image for src tag, it will add a gray border around the image, which is kinda sux. 

Or you want to have full SVG placeholder than to remove it and add image?

File: addon/mixins/in-viewport.js [hunk 4]
```diff
@@ -3,53 +3,62 @@
...
-      rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
-      rect.right <= (window.innerWidth || document.documentElement.clientWidth)
-    );
+      (rect.bottom - get(this, 'threshold')) <= (window.innerHeight || document.documentElement.clientHeight) &&
+      rect.right <= (window.innerWidth || document.documentElement.clientWidth);
+
+    this.set('enteredViewport', enteredViewport);
+
+    if (enteredViewport) {
+      this._unbindScroll();
```
Comment (RuslanZavacky) [inline addon/mixins/in-viewport.js]: This part, by the way, is performance improvement. We unbind scroll event when image is already loaded, as if you have 40 images on the pages, each image continue to listen for the scroll event.

File: addon/components/lazy-image.js (hunk 1 above)
Comment (twokul) [inline addon/components/lazy-image.js]: to avoid using assets, I added [this](https://github.com/twokul/ember-lazy-image/blob/master/app/templates/components/lazy-image.hbs#L4). It works fine for modern browsers. If there's a need to support IE, you can override with your own placeholder.

File: addon/mixins/in-viewport.js (hunk 3 above)
Comment (twokul) [inline addon/mixins/in-viewport.js]: I'd rather keep them indented.

File: addon/mixins/in-viewport.js (hunk 4 above)
Comment (twokul) [inline addon/mixins/in-viewport.js]: do you mind submitting this as a separate PR?

**Comment by twokul** [comment]
seems stale.
